matt/sudoku/2d/utils/board.go

```diff
@@ -215,37 +167,47 @@ func (g *Game) Print() string {
 func (g *Game) ValidGame() bool {
 	for i := 0; i < 9; i++ {
 		for j := 0; j < 9; j++ {
-			if g.gameboard[i][j] == 0 && g.individualSets[i][j] == fullMapping {
+			if g.gameboard[i][j] == 0 && g.valueSetAt(i, j) == fullMapping {
 				return false
 			}
 		}
 	}
 	return true
 }
 
+func (g *Game) valueSetAt(row, col int) valueHitMap {
+	return g.rowsSet[row] | g.colsSet[col] | g.subSquareSets[row/3][col/3]
+}
+
 func (g *Game) NextGuessMove() (row, col int, values []int) {
-	listOfPossibleGuesses := TakenNumbersItems{}
+	var nextGuess *TakenNumbersItem
 
-	for i := 0; i < 9; i++ {
-		for j := 0; j < 9; j++ {
-			if g.individualSets[i][j] != fullMapping && g.gameboard[i][j] == 0 {
-				listOfPossibleGuesses = append(listOfPossibleGuesses, TakenNumbersItem{
-					list: g.individualSets[i][j],
-					row:  i,
-					col:  j,
-				})
+	for c := 0; c < 9; c++ {
+		for r := 0; r < 9; r++ {
+			if g.gameboard[r][c] != 0 {
```

Comment:
Is there a reason you go
```
for c
  for r
```
instead of the inverse? Does that access things in memory in order?

Reply:
I would think the inverse would be more memory efficient. However, found that this ordering solves in fewer tries (I think just a bias in the puzzles selected)